drivers/hmis/lib/tasks/ce_define_workflows.rake

```diff
@@ -106,111 +105,322 @@ module CeWorkflowBuilder
 
+    template = create_template(identifier, template_name, data_source)
+
+    # Create Swimlanes
     ce_staff_swimlane = template.swimlanes.create!(name: 'CE Staff')
     project_staff_swimlane = template.swimlanes.create!(name: 'Project Staff')
 
     start_event = create_start_event(template)
 
-    # Generic form that can be used for many steps. Collects date, notes, and yes/no question
-    # TODO - Update all of this, it's just proof of concept
-    generic_yes_no_form = Hmis::Form::Definition.create!(
-      identifier: GENERIC_YES_NO,
-      status: 'published',
-      title: 'Generic Yes/No Form',
-      role: 'CE_REFERRAL_STEP',
-      version: 0,
+    # Form that is shared across several CE Staff tasks
+    ce_staff_shared_form = {
+      "item": [
+        {
+          "text": 'Date',
+          "type": 'DATE',
+          "link_id": 'date',
+          "required": true,
+          "mapping": { "custom_field_key": 'ce_task_date' },
+        },
+        {
+          "text": 'Notes',
+          "type": 'TEXT',
+          "link_id": 'notes',
+          "required": false,
+          "mapping": { "custom_field_key": 'ce_task_notes' },
+        },
+        {
+          "text": 'Continue with Referral?',
+          "type": 'CHOICE',
+          "link_id": 'move_forward',
+          "required": true,
+          "pick_list_options": [
+            {
+              "code": '1',
+              "label": 'Yes, continue',
+            },
+            {
+              "code": '0',
+              "label": 'No, decline referral',
+            },
+          ],
+          "mapping": { "custom_field_key": 'ce_generic_move_forward' },
+        },
+        {
+          "text": 'Decline Reason',
+          "type": 'CHOICE',
+          "link_id": 'admin_decline_reason',
+          "required": true,
+          "pick_list_options": [
+            { "code": 'HMIS user error' },
+            { "code": 'Client needs to be reassessed' },
+            { "code": 'Does not meet eligibility criteria' },
+            { "code": 'No longer interested in this program' },
+            { "code": 'No longer experiencing homelessness' },
+            { "code": 'Vacancy no longer available' },
+          ],
+          "mapping": { "custom_field_key": 'ac_workflow_v1_admin_decline_reason' },
+          "enable_when": [{ "question": 'move_forward', "operator": 'EQUAL', "answer_code": '0' }],
+        },
+      ],
+    }
+
+    create_step_form(
+      identifier: initial_review_task_form_identifier,
+      definition: ce_staff_shared_form,
+    )
+    create_step_form(
+      identifier: ce_offer_task_form_identifier,
+      definition: ce_staff_shared_form,
+    )
+    create_step_form(
+      identifier: project_offer_task_form_identifier,
+      definition: {
+        "item": [
+          {
+            "text": 'Date',
+            "type": 'DATE',
+            "link_id": 'date',
+            "required": true,
+            "mapping": { "custom_field_key": 'ce_generic_date' },
```

Comment:
Looks like you updated these above to `ce_task_date` and `ce_task_notes`, want to make the same update here?

Reply:
I added a TODO about updating this to generate custom data element definitions; we can handle updating the keys (or having them be generated) there. I'll do that in another PR because I think we want to handle that more generically and make use of the existing code from publishing forms.